book-3-web-api/chapters/CONTROLLER_PARAMETERS.md

```diff
@@ -1,67 +1,91 @@
 
-This kind of parameter is handled in a controller method by specifying the `[FromRoute]` attribute in the argument list. The following method will create a variable `id` and store the value of `1` in it.
+> **NOTE:** ASP<span>.NET</span> Core turns _route parameters_ into _method parameters_ for us. Isn't that nice of it?
 
-```cs
-public async Task<IActionResult> Get([FromRoute]int id)
-```
+In Web API we specify the route parameter in two places. In the HTTP verb attribute (ex. `[HttpGet("{id}")]`) and in the method parameters (ex. `public IActionResult Get(int id)`)
+
+> **NOTE:** Route parameters are not limited to `id`s and can get significantly more complex if the need arises.
 
 ## Query String Parameters
 
-Anything after the route and preceded by a question mark is a query string parameter.
+When working with `json-server` you used `_expand` and `_embed` to optionally include related data.
 
-```sh
-http://localhost:5000/students?cohort=13
+```txt
+http://localhost:3000/animal?_embed=employee
 ```
 
-The `cohort` in the URL above is a query string parameter. It's value is `13`. You can provide as many query string parameters as you want.
+`_expand` and `_embed` are examples of _query string parameters_.
+
+A _query string_ is the part of a URL after the `?`. It consists of a list of `key=value` pairs separated by `&`.
 
-```sh
-http://localhost:5000/students?cohort=13&orderBy=lastname&limit=5
+Consider this URL:
+
+```txt
+https://www.reddit.com/search/?q=csharp&sort=top&t=month
 ```
 
-To get the value of each of those query string parameters, you specify each in the argument list. Any arguments with no attribute in front of them will instruct ASP.NET to look in the query string parameters for the values.
+The query string parameters are `q`, `sort` and `t`. They are used by the Reddit search feature to dictate how the search should be conducted and the search results should be displayed.
 
-> **Note:** The variable names you define in the argument list must match the name in the URL. If you changed `string orderBy` to `string orderColumn`, then it would not capture `lastname` from the URL above.
+* `q` is the search term
+* `sort` defines the order in which the search results should be returned. In this example we want "top" results to be first...whatever that means.
+* `t` specifies the time limit to search. In this example we only want results from the past month
 
-```cs
-public async Task<IActionResult> Get(int? cohort, string orderBy, int limit)
-```
+The Reddit example demonstrates a common use case for query string parameters. They are often used for searching, sorting and configuring how data is returned from a web API.
+
+> **NOTE:** It is important to note that query string parameters can be whatever you, the developer, want them to be. There is nothing special about the query string parameters in the example. They are just what the Reddit developers chose.
```

Comment:
`> **NOTE:** It is important to note that query string parameters can be whatever you, the developer, want them to be. There is nothing special about the query string parameters in the example. They are just what the Reddit developers chose. In the same way, there is nothing special about the _embed or _expand parameters you used with json-server. These are parameter names that the developer of json-server picked out, and are not necessarily relevant keywords outside of json-server`